### DISCUSSION TITLE
How to implement process return or reject in workflow

### DISCUSSION THREAD

**Discussion by landonzeng** [elsa-workflows/elsa-core#2684] [Q&A]
How to achieve the process in workflow to return or refuse to pass, currently are achieved are through, if the rejection of how to return to the previous step?

**Comment by OvyDev** [comment]
You can use `Fork` and `Signal Received` as in the [Document Approval](https://elsa-workflows.github.io/elsa-core/docs/next/guides/guides-document-approval) guide.
You can also use `Switch` or `If/Else` to control flow depending on `input` or some variable.

**Comment by landonzeng** [comment]
> You can use `Fork` and `Signal Received` as in the [Document Approval](https://elsa-workflows.github.io/elsa-core/docs/next/guides/guides-document-approval) guide. You can also use `Switch` or `If/Else` to control flow depending on `input` or some variable.

How to get the headers of the request in signal

**Comment by OvyDev** [comment]
You need to send `workflowInstanceId` or `correlationId` along with the `name` of `signal` to be triggered (use swagger to see the exact parameter). You can also send payload using parameter `input`. The corresponding `Signal Received` (matching `name` and `workflowInstanceId`/`correlationId`) will received sent `input` and passed it along as `output`. 

Your next activity can use this `output` as its `input`, but you need to specify this in the `input` property field (use JavaSrcipt and then type `Input`, it will show up in intelisense). Or in case of custom activity, it will be availabe in `ActivityExecutionContext`.

**Comment by landonzeng** [comment]
> You need to send `workflowInstanceId` or `correlationId` along with the `name` of `signal` to be triggered (use swagger to see the exact parameter). You can also send payload using parameter `input`. The corresponding `Signal Received` (matching `name` and `workflowInstanceId`/`correlationId`) will received sent `input` and passed it along as `output`.
> 
> Your next activity can use this `output` as its `input`, but you need to specify this in the `input` property field (use JavaSrcipt and then type `Input`, it will show up in intelisense). Or in case of custom activity, it will be availabe in `ActivityExecutionContext`.

I want to get the Authorization in the request header in the signal

**Comment by OvyDev** [comment]
Do you mean you need to protect signal endpoint behind authorization ? 
If so, just configure asp net built in authorization.

**Comment by landonzeng** [comment]
> Do you mean you need to protect signal endpoint behind authorization ? If so, just configure asp net built in authorization.

I want to request signal with authentication information from other projects, such as jwt, and after signal receives the information, sendhttpRequest is executed after the request to call the api of other projects.

**Comment by OvyDev** [comment]
I don't think its possible, if your triggering the signal via the `signal_url` as in [Document Approval](https://elsa-workflows.github.io/elsa-core/docs/next/guides/guides-document-approval) guide.

However you can trigger the signal from your own `Controller` by injecting `ISignaler`. You can then send the jwt as `input` to the `Signal Received`. This `input` then flows to the next activity `input`.
